### PR TITLE
feat(behavior_velocity): add velocity smoother to behavior velocity

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Related Issue(required)

#258 
#383

## Description(required)

- [x] add stop line at the end 
move stop line module before occlusion spot here
- [x] add external velocity limit not to exceed limit velocity 
https://github.com/tier4/autoware_launch/pull/238
- [x] ttc calculation and enable pass judge
use velocity smoother in this PR

smooth method is below
1. path to traj
2. external_v_limit (default) 
3. lateral acc filter
4. lerp ego cloesest
5. resample
6. smooth
7. concat
8. traj to path

pass_judge option is going to be enable for this PR  "smooth_velocity" is available
```
      detection_method: "occupancy_grid" # [-] candidate is "occupancy_grid" or "predicted_object"
      pass_judge: "current_velocity"       # [-] candidate is "current_velocity" or "smooth_velocity"
```

## Review Procedure(required)

- test in experiment at kashiwanoha

## Review Procedure(required)

- launch psim
- test using real vehicle is necessary

with #497

https://user-images.githubusercontent.com/65527974/158551188-468312cc-6472-42c1-9a3e-a6e27d51578b.mp4

https://user-images.githubusercontent.com/65527974/158551202-2fae6da4-7c46-4346-b4e0-bc4861e372bd.mp4

## Related PR(optional)

FYI: @TakaHoribe 

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: planning/behavior_velocity_planner/src/node.cpp [hunk 1]
```diff
@@ -162,6 +163,10 @@ BehaviorVelocityPlannerNode::BehaviorVelocityPlannerNode(const rclcpp::NodeOptio
   if (this->declare_parameter("launch_stop_line", true)) {
     planner_manager_.launchSceneModule(std::make_shared<StopLineModuleManager>(*this));
   }
+  // to calculate ttc it's better to be after stop line
+  if (this->declare_parameter("launch_occlusion_spot", true)) {
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/node.cpp:167]: move module order to insert stop line at the end before calculation of ttc.

**Comment by taikitanaka3** [comment]
[7cd692a](https://github.com/autowarefoundation/autoware.universe/pull/497/commits/7cd692a2f1d6016760ea3be9bb4ab3e15ff9a7c5)

until is previous PR of refactoring.https://github.com/autowarefoundation/autoware.universe/pull/528

File: planning/behavior_velocity_planner/src/node.cpp [hunk 2]
```diff
@@ -283,6 +292,66 @@ void BehaviorVelocityPlannerNode::onVehicleVelocity(
   }
 }
 
+void BehaviorVelocityPlannerNode::onParam()
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/node.cpp:295]: @mkuri -san is going to replace this with loadParam inside motion velocity smoother

Comment (mkuri) [reply planning/behavior_velocity_planner/src/node.cpp:298]: 想定通りにコンストラクトしてもらっています。

**Review by mkuri — APPROVED** [review]
smootherの各メンバ関数への入力や順番は検証済みとして、使い方やコンストラクトの方法を確認しました。
LGTM

**Pull request merged**